Support subplot creation at runtime via `createGrid`

Vindaar opened the pull request:
This PR adds a `Grid` object and corresponding `createGrid` procedure, which can be used to create subplots using a grid at runtime. It's basically what was proposed in #43. 

Plots can be assigned to the `Grid` object using `[]=`. In addition more plots than defined by the `numPlots` argument of `createGrid` can be added by using `add`. 
The given object is converted to `PlotJson`. A call to `show` on the grid, will calculate the required number of rows and columns (`numPlotsPerRow` is an optional argument to `createGrid`) and make a call to `combine` to get a valid `PlotJson` object of the subplot.

I had to move the implementation from `plotly.nim` to a submodule, to be able to call `show` from within `plotly_subplots`. 

The `Grid` object has a `layout` field, which takes just a normal `Layout` object. This can also be given to `createGrid`, but can be set later if desired.

For some reason the `show` proc with `filename` arg for `--threads:off` does not compile. It acts as if the proc was called, even it isn't. Not sure what's up there.

I'm all ears for other ideas, features of different names for the args / objects.

@timotheecour let me know if this suits your needs. :)

LemonBoy commented on examples/fig18_subplots.nim:
```diff
@@ -101,3 +101,21 @@ let pltC2 = subplots:
+# Note: internally the returned `Grid` object stores all plots already
+# converted to `PlotJson` (i.e. the `layout` and `traces` fields are
+# `JsonNodes`).
+var grid = createGrid(numPlots = 2) #,
```
I'd prefer a `rows, columns: int` and a `[]`/`[]=` that takes a row/column pair in order to better work with MxN subplots

Vindaar replied:
I personally agree with you. That's what I intuitively would expect when dealing with a 2D grid anyways. @timotheecour your opinion?
I'm fine either way though.

brentp commented:
I am fine with this. Seems like soon we need more general documentation now that the feature-set is expanding.

Vindaar commented:
> I am fine with this. Seems like soon we need more general documentation now that the feature-set is expanding.

Yes, I agree. I've been meaning to write some more documentation... :|

Vindaar commented:
Sorry for taking so long to get back to this.
I just:
- added the option to assign subplots via `(row, col)` coordinate tuples
- fixed a bug in the row / col calculation if `numPlotsPerRow` was set
- renamed and exported `showImpl` to `toPlotJson`
- fixed access to the JS backend

brentp commented:
I haven't had a look at this. @LemonBoy PTAL?

@Vindaar is this still as you'd like it?

Vindaar commented:
For me this is fine. I think it's ok to support both ways of adding elements to the grid.
I've been using it in my own code already without issues.
